src/lib.rs

```diff
@@ -139,9 +141,12 @@ mod contract {
     pub extern "C" fn deploy_code() {
         let input = sdk::read_input();
         let mut engine = Engine::new(predecessor_address());
-        let (status, address) = Engine::deploy_code_with_input(&mut engine, &input);
+        let (status, result) = Engine::deploy_code_with_input(&mut engine, &input);
         // TODO: charge for storage
-        process_exit_reason(status, &address.0)
+        process_exit_reason(
+            status,
+            &result.try_to_vec().expect("serialize can not fail"),
```

Comment:
I think we should try to move away from the built-in `expect`/`unwrap` calls on `Result` and `Option` in favour of ones that use the `sdk::panic` call.

It doesn't make a big difference right now because our binary is largely bloated by secp256 data, but if we want to optimize code size in the future then avoiding the rust panic is a good step.

Also, in the spirit of https://github.com/aurora-is-near/aurora-engine/pull/42#discussion_r620259522 this should probably be something like `ERR_SERIALIZE` or something like that. And it should probably be factored out as a constant since it is repeated in multiple places.

Reply:
Sure, it shouldn't be able to fail regardless, hence the expect but, I can change it to a `sdk_unwrap` as a "just in case" kind of thing.

Reply:
Second though, I can't make this into a `sdk_unwrap` as expected as the `Error` returned doesn't implement `ToStr`.

What I would like to do instead is to define our own `ExitError` which may be 1:1, or nearly, what we are currently using, then implement `Display` on that which would then be compatible with every error for `sdk_unwrap()`. I'd say, lets go with it as is for now, and then I'll do a separate PR where I would do `sdk_unwrap()` with the new `Error` type that implements `Display`. Unusually, EVM `ExitError` _-doesn't-_ implement `Display` which I would consider a requirement for any kind of Error to implement both `Debug` and `Display`.

Reply:
Addressed by https://github.com/aurora-is-near/aurora-engine/pull/49?

Reply:
Yes, this is addressed by #49 which is in the description of it.